dstep/translator/IncludeHandler.d

```diff
@@ -109,21 +123,43 @@ class IncludeHandler
         imports ~= "core.stdc.config";
     }
 
+    /// Makes includes that match regex filter be converted to import with prefix.
+    void setAutoImportPrefix (string prefix)
+    {
+        this.convertIncludes = true;
+        this.importPrefix = prefix;
+    }
+
+    /// Makes includes that match regex filter be converted to import with prefix.
+    void setAutoImportFilter (string filter)
+    {
+        this.convertIncludes = true;
+        this.convertableIncludePattern = regex(filter);
+    }
+
     string[] toImports ()
     {
-        auto r =  rawIncludes.map!((e) {
+        auto r =  mambo.core.Array.map!((e) {
             if (auto i = isKnownInclude(e))
                 return toImport(i);
-
+            else if( this.convertIncludes && isConvertableInclude(e) )
+                return toImport(autoConvertInclude(e));
             else
                 return "";
-        });
+        })(rawIncludes);
 
-        auto imps = imports.map!(e => toImport(e));
+        auto imps = mambo.core.Array.map!(e => toImport(e))(imports);
 
         return r.append(imps).filter!(e => e.any).unique.toArray;
     }
 
+    /// Returns the base name (last component without extension) of a file path.
+    static string baseName (string path)
+    {
+        string last_component = text(retro(Path.pathSplitter(path)).front);
+        return Path.stripExtension( last_component );
+    }
```

Comment:
Phobos already contains `baseName` [1], would that work?

[1] http://dlang.org/phobos/std_path.html#.baseName